To: vim_dev@googlegroups.com
Subject: Patch 7.3.316
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.316 (after 7.3.306)
Problem:    Crash when 'colorcolumn' is set and closing buffer.
Solution:   Check for w_buffer to be NULL. (Yasuhiro Matsumoto)
Files:	    src/option.c


*** ../vim-7.3.315/src/option.c	2011-06-12 22:13:37.000000000 +0200
--- src/option.c	2011-09-21 13:26:31.000000000 +0200
***************
*** 7036,7041 ****
--- 7036,7044 ----
      int		i;
      int		j = 0;
  
+     if (wp->w_buffer == NULL)
+ 	return NULL;  /* buffer was closed */
+ 
      for (s = wp->w_p_cc; *s != NUL && count < 255;)
      {
  	if (*s == '-' || *s == '+')
*** ../vim-7.3.315/src/version.c	2011-09-14 19:04:35.000000000 +0200
--- src/version.c	2011-09-21 13:40:05.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     316,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
25. You believe nothing looks sexier than a man in boxer shorts illuminated
    only by a 17" inch svga monitor.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
